src/Tests/Tests.MonoTorrent.Client/MonoTorrent/TorrentV2Test.cs

```diff
@@ -116,6 +116,7 @@ public void LoadV2OnlyTorrent ()
             Assert.AreEqual (InfoHash.FromHex ("caf1e1c30e81cb361b9ee167c4aa64228a7fa4fa9f6105232b28ad099f3a302e"), V2OnlyTorrent.InfoHashes.V2);
         }
 
+        // TODO: fails with MonoTorrent.TorrentException : the 'piece layers' dictionary did not contain an entry for the file 'tbl-tint.mpg'
```

Comment:
I pushed a tiny change which fixes this. When zeroing out the padding for the last file in the torrent you had forgotten to copy over the `RootHash` property. This is what caused the file to not be found in the piece layers dictionary :)